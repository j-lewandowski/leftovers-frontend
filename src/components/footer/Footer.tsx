import { Divider, Stack, styled, Typography } from '@mui/material';
import facebookLogo from '../../assets/facebook.png';
import instagramLogo from '../../assets/instagram.png';

const Footer = () => {
  return (
    <FooterWrapper>
      <ContentGroup
        gap={{ xs: 0, sm: 4 }}
        direction={{ xs: 'column', sm: 'row' }}
      >
        <ContentGroup gap={{ xs: 2, sm: 0.5 }} direction="row">
          <Link href="https://www.instagram.com/">
            <ImageLogo src={instagramLogo} alt="instagram logo" />
          </Link>
          <Link href="https://www.facebook.com/">
            <ImageLogo src={facebookLogo} alt="facebook logo" />
          </Link>
        </ContentGroup>

        <ContentGroup
          direction="row"
          divider={<Divider orientation="vertical" flexItem />}
          gap={1}
        >
          <Body3TextSecondary>contact@leftovers.com</Body3TextSecondary>

          <Link href="#">
            <Body3TextPrimary>Terms of Services</Body3TextPrimary>
          </Link>
          <Link href="#">
            <Body3TextPrimary>Privacy Policy</Body3TextPrimary>
          </Link>
        </ContentGroup>
      </ContentGroup>

      <Body3TextSecondary>
        powered by <BotAI>BotAI</BotAI>
      </Body3TextSecondary>
    </FooterWrapper>
  );
};

const FooterWrapper = styled('footer')(({ theme }) => ({
  width: '100%',
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  [theme.breakpoints.down('sm')]: {
    flexDirection: 'column',
    padding: '1rem',
  },
  flexDirection: 'row',
  padding: '1rem 4.5rem',
}));

const ImageLogo = styled('img')({
  height: '1.5rem',
});

const ContentGroup = styled(Stack)({
  alignItems: 'center',
});

const Body3TextSecondary = styled(Typography)(({ theme }) => ({
  fontWeight: '400',
  color: theme.palette.text.secondary,
  fontSize: '12px',
}));

const Body3TextPrimary = styled(Typography)(({ theme }) => ({
  fontWeight: '400',
  color: theme.palette.text.primary,
  fontSize: '12px',
}));

const Link = styled('a')({
  textDecoration: 'none',
});

const BotAI = styled('span')({
  textDecoration: 'underline',
});

export default Footer;
