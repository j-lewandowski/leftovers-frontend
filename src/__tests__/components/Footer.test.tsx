import { render } from '@testing-library/react';
import Footer from '../../components/footer/Footer';
import { describe, test, expect } from 'vitest';

describe('Footer', () => {
  test('renders footer', () => {
    const result = render(<Footer />);
    expect(result).toMatchSnapshot();
  });
});
