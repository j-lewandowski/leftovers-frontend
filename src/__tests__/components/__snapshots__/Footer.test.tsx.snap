// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > renders footer 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="css-txnbr9"
      >
        <div
          class="MuiStack-root css-1r16e-MuiStack-root"
        >
          <div
            class="MuiStack-root css-1ralu9w-MuiStack-root"
          >
            <a
              class="css-ytumd6"
              href="https://www.instagram.com/"
            >
              <img
                alt="instagram logo"
                class="css-1f1jesu"
                src="/src/assets/instagram.png"
              />
            </a>
            <a
              class="css-ytumd6"
              href="https://www.facebook.com/"
            >
              <img
                alt="facebook logo"
                class="css-1f1jesu"
                src="/src/assets/facebook.png"
              />
            </a>
          </div>
          <div
            class="MuiStack-root css-1rf2zra-MuiStack-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-11x867o-MuiTypography-root"
            >
              contact@leftovers.com
            </p>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
            />
            <a
              class="css-ytumd6"
              href="#"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1u0dmiv-MuiTypography-root"
              >
                Terms of Services
              </p>
            </a>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
            />
            <a
              class="css-ytumd6"
              href="#"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1u0dmiv-MuiTypography-root"
              >
                Privacy Policy
              </p>
            </a>
          </div>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-11x867o-MuiTypography-root"
        >
          powered by 
          <span
            class="css-1081t4c"
          >
            BotAI
          </span>
        </p>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="css-txnbr9"
    >
      <div
        class="MuiStack-root css-1r16e-MuiStack-root"
      >
        <div
          class="MuiStack-root css-1ralu9w-MuiStack-root"
        >
          <a
            class="css-ytumd6"
            href="https://www.instagram.com/"
          >
            <img
              alt="instagram logo"
              class="css-1f1jesu"
              src="/src/assets/instagram.png"
            />
          </a>
          <a
            class="css-ytumd6"
            href="https://www.facebook.com/"
          >
            <img
              alt="facebook logo"
              class="css-1f1jesu"
              src="/src/assets/facebook.png"
            />
          </a>
        </div>
        <div
          class="MuiStack-root css-1rf2zra-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-11x867o-MuiTypography-root"
          >
            contact@leftovers.com
          </p>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
          />
          <a
            class="css-ytumd6"
            href="#"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1u0dmiv-MuiTypography-root"
            >
              Terms of Services
            </p>
          </a>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
          />
          <a
            class="css-ytumd6"
            href="#"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1u0dmiv-MuiTypography-root"
            >
              Privacy Policy
            </p>
          </a>
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-11x867o-MuiTypography-root"
      >
        powered by 
        <span
          class="css-1081t4c"
        >
          BotAI
        </span>
      </p>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
